### PR TITLE
Require NsatzTactic: nsatz support for Z and Q

### LINKED ISSUE #12860: Require Import NsatzTactic should allow nsatz to work over Z and Q
The purpose of NsatzTactic is to allow using `nsatz` without the dependency on real axioms.  So we shouldn't have to re-create instances manually for `Z` and `Q`.

### PR DESCRIPTION
**Pull request by JasonGross**
The purpose of `NsatzTactic` is to allow using `nsatz` without the
dependency on real axioms. So we declare the instances for `Z` and `Q`
in that file, so that users don't have to re-create them.

Fixes #12860

**Kind:** bug fix / enhancement

Fixes / closes #12860

- [x] Added / updated test-suite
- Corresponding documentation was added / updated (there doesn't seem to be any doc to update).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Review by thery — COMMENTED** [review]
Looks good but maybe there should be a footnote in the doc https://coq.inria.fr/refman/addendum/nsatz.html that tells what is going on.

**Comment by JasonGross** [comment]
Documentation updated; is this approximately what you were thinking of?

File: doc/changelog/10-standard-library/12861-nsatz-tactic-instances.rst
Comment (jfehrle) [inline doc/changelog/10-standard-library/12861-nsatz-tactic-instances.rst:5]:
```suggestion
  ``Require Import Coq.nsatz.NsatzTactic`` now allows using :tacn:`nsatz`
  with `Z` and `Q` without having to supply instances or using ``Require Import Coq.nsatz.Nsatz``, which
  transitively requires unneeded files declaring axioms used in the reals
```

If that doesn't change the meaning.

File: doc/sphinx/addendum/nsatz.rst
Comment (jfehrle) [inline doc/sphinx/addendum/nsatz.rst:41]:
```suggestion
   Alternatively, if you prefer not to transitively depend on the
   files declaring the axioms used to define the real numbers, you can
   ``Require Import NsatzTactic`` instead; this will still allow
   :tacn:`nsatz` to solve goals defined about :math:`\mathbb{Z}`,
   :math:`\mathbb{Q}` and any user-registered rings.
```

Is there a simpler phrase than "declaring the axioms used to define"?

**Review by thery — APPROVED** [review]
fine with me

File: doc/sphinx/addendum/nsatz.rst
Comment (jfehrle) [inline doc/sphinx/addendum/nsatz.rst:94]: Not so convenient for users who haven't downloaded the source tree.  Maybe put in a hyperlink to the file in GitHub?

**Review by jfehrle — APPROVED** [review]
Only looked at the doc, a few wording suggestions, please change the ones that make sense to you.

File: doc/sphinx/addendum/nsatz.rst
Comment (JasonGross) [reply doc/sphinx/addendum/nsatz.rst:94]: cc @coq/doc-maintainers , how should I hyperlink to test-suite files?  (and is there a way to version it, or should I just link to current master?)

File: doc/sphinx/addendum/nsatz.rst
Comment (JasonGross) [reply doc/sphinx/addendum/nsatz.rst:41]: Not that I'm aware of

File: doc/sphinx/addendum/nsatz.rst
Comment (jfehrle) [reply doc/sphinx/addendum/nsatz.rst:94]: The markup syntax is like this, in which `Ocaml Gc` is the displayed text (e.g. `Nsatz.v` in your case):

```
`OCaml Gc <http://caml.inria.fr/pub/docs/manual-ocaml/libref/Gc.html#VALcompact>`_
```

Even if GitHub allows linking to a specific version of the file (I'm guessing it does), we don't yet have a mechanism/procedure to update such links in future releases of Coq.  Maybe we should create that.  Not sure how many places in the doc we would use that feature.  @Zimmi48, your thoughts?

The words "the file" seem unnecessary, BTW.

Comment (JasonGross) [reply doc/sphinx/addendum/nsatz.rst:94]: @jfehrle So like this?
```suggestion
See `Nsatz.v in the test-suite <https://github.com/coq/coq/blob/master/test-suite/success/Nsatz.v>`_ for many examples, especially in geometry.
```
Is there a way to get `Nsatz.v` to still be teletype font even as part of a link?

Comment (jfehrle) [reply doc/sphinx/addendum/nsatz.rst:94]: > Is there a way to get `Nsatz.v` to still be teletype font even as part of a link?

Not that I'm aware of.  My impression is that a lot of the markup operations are not orthogonal and can't be combined.

Maybe move "in the test-suite" so it's after and not part of the link.

Comment (Zimmi48) [reply doc/sphinx/addendum/nsatz.rst:94]: I think linking to master is fine. We don't have that many links so far.

Comment (JasonGross) [reply doc/sphinx/addendum/nsatz.rst:94]:
```suggestion
See the test-suite file `Nsatz.v <https://github.com/coq/coq/blob/master/test-suite/success/Nsatz.v>`_ for many examples, especially in geometry.
```

**Comment by SkySkimmer** [comment]
@JasonGross this should probably be squashed
@coq/nsatz-maintainers please assign

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Squashing the documentation commits looks like a good idea indeed, but it's not mandatory either. I'll merge soon anyways.

**Comment by jfehrle** [comment]
Please squash before the merge.  The individual commits are not interesting, more noise in the git history.

**Comment by JasonGross** [comment]
I've squashed all the commits.

**Pull request merged**
